reasoner/resolution/ResolverRegistry.java

```diff
@@ -208,16 +216,44 @@ public ResolverView registerResolvable(Resolvable<?> resolvable) {
     public Actor.Driver<BoundConcludableResolver> registerBoundConcludable(
             Concludable concludable, ConceptMap bounds, Map<Actor.Driver<ConclusionResolver>, Rule> resolverRules,
             LinkedHashMap<Actor.Driver<ConclusionResolver>, Set<Unifier>> applicableRules,
-            Actor.Driver<? extends Resolver<?>> root, int iteration) {
+            Actor.Driver<? extends Resolver<?>> root, int iteration, boolean explain) {
         LOG.debug("Register BoundConcludableResolver, pattern: {} bounds: {}", concludable.pattern(), bounds);
-        Actor.Driver<BoundConcludableResolver> resolver = Actor.driver(driver -> new BoundConcludableResolver(
-                driver, concludable, bounds, resolverRules, applicableRules, this, traversalEngine, conceptMgr, resolutionTracing
-        ), executorService);
+        Actor.Driver<BoundConcludableResolver> resolver;
+        if (explain) {
+            resolver = Actor.driver(driver -> new ExplainBoundConcludableResolver(
+                    driver, concludable, bounds, resolverRules, applicableRules, this, traversalEngine, conceptMgr,
+                    resolutionTracing), executorService);
+        } else {
+            resolver = Actor.driver(driver -> new MatchBoundConcludableResolver(
+                    driver, concludable, bounds, resolverRules, applicableRules, this, traversalEngine, conceptMgr,
+                    resolutionTracing), executorService);
+        }
         resolvers.add(resolver);
         reiterationQueryRespondents.computeIfAbsent(new Pair<>(root, iteration), r -> new HashSet<>()).add(resolver);
         if (terminated.get()) throw TypeDBException.of(RESOLUTION_TERMINATED); // guard races without synchronized
         return resolver;
     }
+
+    public Actor.Driver<BoundConclusionResolver> registerBoundConclusion(
+            Rule.Conclusion conclusion, ConceptMap bounds, Actor.Driver<ConditionResolver> conditionResolver,
+            boolean explain) {
+        LOG.debug("Register BoundConclusionResolver, pattern: {} bounds: {}", conclusion.conjunction(), bounds);
+        Actor.Driver<BoundConclusionResolver> resolver;
+        if (explain) {
+            // TODO: Use explain resolver
+            resolver = Actor.driver(driver -> new BoundConclusionResolver(
+                    driver, conclusion, bounds, conditionResolver, materialiser, this, traversalEngine, conceptMgr,
+                    resolutionTracing), executorService);
+        } else {
+            // TODO: Use match resolver
+            resolver = Actor.driver(driver -> new BoundConclusionResolver(
+                    driver, conclusion, bounds, conditionResolver, materialiser, this, traversalEngine, conceptMgr,
+                    resolutionTracing), executorService);
```

Comment:
We could discuss this - it's ready to split the Match and Explain, but actually for conclusion rather than concludable the code paths don't justify a whole new class to manage each possibility. They're basically one method different.